lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -42,18 +44,44 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(const UsdSceneItem::Ptr& srcItem, con
 	: Ufe::UndoableCommand()
 {
 	const UsdPrim& prim = srcItem->prim();
-	fStage = prim.GetStage();
-	fUfeSrcItem = srcItem;
-	fUsdSrcPath = prim.GetPath();
-	// Every call to rename() (through execute(), undo() or redo()) removes
+	_stage = prim.GetStage();
+	_ufeSrcItem = srcItem;
+	_usdSrcPath = prim.GetPath();
+
+    // Every call to rename() (through execute(), undo() or redo()) removes
 	// a prim, which becomes expired.  Since USD UFE scene items contain a
 	// prim, we must recreate them after every call to rename.
-	fUsdDstPath = prim.GetParent().GetPath().AppendChild(TfToken(newName.string()));
-	fLayer = defPrimSpecLayer(prim);
-	if (!fLayer) {
+	_usdDstPath = prim.GetParent().GetPath().AppendChild(TfToken(newName.string()));
+	_layer = MayaUsdUtils::defPrimSpecLayer(prim);
+	if (!_layer) {
 		std::string err = TfStringPrintf("No prim found at %s", prim.GetPath().GetString().c_str());
 		throw std::runtime_error(err.c_str());
 	}
+
+    // if the current layer doesn't have any opinions that affects selected prim
+    if (!MayaUsdUtils::doesLayerHavePrimSpec(prim)) {
+        auto possibleTargetLayer = MayaUsdUtils::strongestLayerWithPrimSpec(prim);
+        std::string err = TfStringPrintf("Cannot rename [%s] defined on another layer. " 
+                                         "Please set [%s] as the target layer to proceed", 
+                                         prim.GetName().GetString(), 
+                                         possibleTargetLayer->GetDisplayName());
+        throw std::runtime_error(err.c_str());
+    }
+    else
+    {
+        auto layers = MayaUsdUtils::layersWithOpinion(prim);
+
+        if (layers.size() > 1) {
+            std::string layerNames;
+            for (auto layer : layers) {
+                layerNames.append("[" + layer->GetDisplayName() + "]" + ",");
+            }
+            layerNames.pop_back();
```

Comment:
Why omit the last layer?

Reply:
see 4c67b698066f1e18b8c40caf6e3a1158066c52d2

I rename `layerNames` to `layerDisplayNames `so it reads better.

I am not omitting any layers and rather the extra "," at the end of `layerDisplayNames ` string.

This is the message the design wants to show up as part of the error message:

`Opinions exist in [layer name 1], [layer name 2], [layer name 3], [layer name 4].”`

So when I iterate through the layers, I get their GetDisplayName and concatenate them in the
layerDisplayNames string separated by ","

```
for (auto layer : layers) {
    layerDisplayNames.append("[" + layer->GetDisplayName() + "]" + ",");
}
layerDisplayNames.pop_back();
```
There are better ways of doing this, but this is one way I did.